File: app/src/main/java/com/arash/altafi/volley/java/adapter/NewsAdapter.java
package com.arash.altafi.volley.java.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.arash.altafi.volley.R;
import com.arash.altafi.volley.java.data.ResponseNews;
import com.bumptech.glide.Glide;
import com.google.android.material.card.MaterialCardView;
import java.util.ArrayList;
import java.util.List;

import de.hdodenhof.circleimageview.CircleImageView;

public class NewsAdapter extends RecyclerView.Adapter<NewsAdapter.NewsViewHolder> {

    List<ResponseNews> responseNews = new ArrayList<>();
    Context context;

    public NewsAdapter(List<ResponseNews> responseNews, Context context) {
        this.responseNews = responseNews;
        this.context = context;
    }

    @NonNull
    @Override
    public NewsViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_news,parent,false);
        return new NewsViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull NewsViewHolder holder, int position) {
        ResponseNews news = responseNews.get(position);
        holder.txtTitle.setText(news.getTitle());
        holder.txtContent.setText(news.getContent());
        holder.txtAuthor.setText(news.getName());
        holder.txtDate.setText(news.getDate());
        Glide.with(context).load(news.getImage()).into(holder.imgNews);
        Glide.with(context).load(news.getWriterImage()).into(holder.imgAuthor);
    }

    @Override
    public int getItemCount() {
        return responseNews.size();
    }

    public class NewsViewHolder extends RecyclerView.ViewHolder {

        private TextView txtContent,txtTitle,txtAuthor,txtDate;
        private CircleImageView imgAuthor;
        private ImageView imgNews;

        public NewsViewHolder(@NonNull View itemView) {
            super(itemView);
            txtDate = itemView.findViewById(R.id.txt_date_news);
            txtAuthor = itemView.findViewById(R.id.txt_author_news);
            txtContent = itemView.findViewById(R.id.txt_content_news);
            txtTitle = itemView.findViewById(R.id.txt_title_news);
            imgAuthor = itemView.findViewById(R.id.img_author_news);
            imgNews = itemView.findViewById(R.id.img_news);
        }
    }

}

File: app/src/main/java/com/arash/altafi/volley/kotlin/adapter/NewsAdapter.kt
package com.arash.altafi.volley.kotlin.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.arash.altafi.volley.R
import com.arash.altafi.volley.kotlin.data.ResponseNewsKotlin
import com.bumptech.glide.Glide
import de.hdodenhof.circleimageview.CircleImageView

class NewsAdapter(private val newsList : List<ResponseNewsKotlin>) : RecyclerView.Adapter<NewsViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): NewsViewHolder {
        val view : View = LayoutInflater.from(parent.context).inflate(R.layout.item_category , parent , false)
        return NewsViewHolder(view)
    }

    override fun onBindViewHolder(holder: NewsViewHolder, position: Int) {
        val list = newsList[position]
        holder.txtTitle.text = list.title
        holder.txtContent.text = list.content
        holder.txtAuthor.text = list.name
        holder.txtDate.text = list.date
        Glide.with(holder.itemView.context).load(list.image).into(holder.imgNews)
        Glide.with(holder.itemView.context).load(list.writerImage).into(holder.imgAuthor)
    }

    override fun getItemCount(): Int = newsList.size

}

class NewsViewHolder(itemView : View) : RecyclerView.ViewHolder(itemView) {

    var txtContent : TextView = itemView.findViewById(R.id.txt_content_news)
    var txtTitle :TextView = itemView.findViewById(R.id.txt_title_news)
    var txtAuthor :TextView = itemView.findViewById(R.id.txt_author_news)
    var txtDate :TextView = itemView.findViewById(R.id.txt_date_news)
    val imgAuthor : CircleImageView = itemView.findViewById(R.id.img_author_news)
    val imgNews: ImageView = itemView.findViewById(R.id.img_news)

}